### ISSUE TITLE
Onboarding Screen

### ISSUE THREAD

**Issue by snuff4** [openfoodfacts/openfoodfacts-ios#89]
Present in the native Android application, ​a series of screens to introduce the features of Open Food Facts to new users ​after the Launch Screen.

❏ This will be present only the first time the User opens the application.
❏ An option to skip to the Main screen, and the last onboarding screen will have a button "Get Started". 
❏ These screens will consist of an image, small amount of text and navigation between pages.

**Comment by snuff4** [comment]
Thank you for your feedback! I hope from modularity you mean that all four versions would share the same code, with only some differences; and apps' app-icons and splash images are different. 

Like we created flavours in Android, we can start with creating Xcode Targets. 

What do you think  @teolemon ?

**Comment by teolemon** [comment]
Yes, exactly. @roger-tan is more the expert here, but that's the idea.

**Comment by snuff4** [comment]
@teolemon Unfortunate that I was not selected, maybe I was not good enough. But I really worked hard like anything on the proposal. But I will still contribute and try to complete. Is it okay @teolemon ?

**Comment by roger-tan** [comment]
Sorry for late answer. I don't have an Android Phone right now. But I'm guessing. We will probably need to create a OnboardingViewController that contains an ImageView/Lottie with a label inside a stackView. Later, we can set this controller with different parameters. I would not recommend to use a storyboard if we make easy to reusable. Then we can use UIPageController to handle views and indicator. Let me check when I have a free time to check how Android Team are doing their onboarding. (If someone has an Android Phone, could I have a screencast to ensure which direction we should go otherwise. I will try to find a free time to check how to architect this).

**Comment by snuff4** [comment]
Thank you for your response @roger-tan. I thought on the same line as yours. I was inspired by [this](https://github.com/Ramotion/paper-onboarding).
I have an Android Phone, and I will send a screencast asap.

**Comment by roger-tan** [comment]
I think the architecture will an Onboarding Page Controller or A scrollview of a horizontal stack view which contains the same children which has a stackView which contain a imageView, two labels. 

Let's define together how we can do an on-boarding without extra framework.

**Comment by aleene** [comment]
So this implies that you need to copy all existing views and make it part of onboarding. And embed these in the pages of a pageViewController. What when you want to present logic of the views.
(in other words, I have no idea how this should work. Seems a lot of work.)

**Comment by teolemon** [comment]
Unpopular opinion, use a library ?
https://medium.com/better-programming/5-swift-libraries-for-designing-a-beautiful-onboarding-experience-2075d05924ea

**Comment by teolemon** [comment]
We have started using Bulletin board already.
